themes.go

```diff
@@ -0,0 +1,98 @@
+//go:generate sh sync-themes.sh
+package main
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+)
+
+var (
+	//go:embed themes.json
+	themesBts []byte
+
+	//go:embed themes_custom.json
+	customThemesBts []byte
+
+	themesOnce sync.Once
+	allThemes  = map[string]Theme{}
+)
+
+func sortedThemeNames() []string {
+	keys := make([]string, 0, len(Themes()))
+	for k := range Themes() {
+		keys = append(keys, k)
+	}
+	sort.Slice(keys, func(i, j int) bool {
+		return strings.ToLower(keys[i]) < strings.ToLower(keys[j])
+	})
+	return keys
+}
+
+// Themes return the list of themes.
+func Themes() map[string]Theme {
+	themesOnce.Do(func() {
+		var themes []windowsTerminalTheme
+		for _, bts := range [][]byte{themesBts, customThemesBts} {
+			if err := json.Unmarshal(bts, &themes); err != nil {
+				fmt.Fprintf(os.Stderr, "could not load themes.json: %s\n", err)
+				os.Exit(1)
+			}
+			for _, theme := range themes {
```

Comment:
How come we need to do this for all the themes?
Each tape file can use only one theme. It feels like a lot of memory going to be used to do this conversion for only using one theme.
Perhaps we can keep all the themes just how they are (with purple and stuff) and then convert when we want to use it? Just an idea, though.

Reply:
One thought I had was to actually filter and load just the one theme asked... which seems more reasonable actually, will do that I think

Reply:
Sounds great! Thank you